internal/handlers/pg/msg_find.go

```diff
@@ -154,3 +169,65 @@ func (h *Handler) MsgFind(ctx context.Context, msg *wire.OpMsg) (*wire.OpMsg, er
 
 	return &reply, nil
 }
+
+func getMaxTimeMSParameter(document *types.Document) (int32, error) {
+	var maxTimeMS int32
+	maxTimeMSParam, err := common.GetOptionalParam(document, "maxTimeMS", maxTimeMS)
+	if err != nil {
+		var commonErr *common.Error
+		if errors.As(err, &commonErr) {
+			return 0, getErrorForInvalidMaxTimeMS(document)
+		}
+
+		return 0, err
+	}
+
+	if maxTimeMSParam < 0 {
+		return 0, common.NewErrorMsg(
+			common.ErrBadValue,
+			fmt.Sprintf("%v value for maxTimeMS is out of range", maxTimeMSParam),
```

Comment:
"Out of range" doesn't sound specific enough to me. I'd suggest something like `maxTimeMS must not be negative (but %v is provided)`

Reply:
As i mentioned in the comment below, I tried to make the answers as much as possible the same as in mongodb, should I change this to what you suggested or can we leave the answer that would be returned when answering with mongodb?